src/js/mixins/WindowListenerMixin.js

```diff
@@ -0,0 +1,20 @@
+import { each } from 'lodash';
+import $ from 'jQuery';
+
+const $window = $(window);
+
+export default {
+  componentDidMount() {
+    this._windowListeners = this.getWindowListeners()
+    each(this._windowListeners, (handler, eventName) => {
+      $window.on(eventName, handler);
+    });
+  },
+
+  componentWillUnmount() {
```

Comment:
'concise methods' is available in ES6 (use esnext option) or Mozilla JS extensions (use moz).